### PR TITLE
Feature/ bag support

### PR DESCRIPTION
**Pull request by shino**
To enable multibag feature.

### CODE REVIEW

File: rebar.config.script [hunk 1]
```diff
@@ -61,7 +90,7 @@ case UseEEDeps of
 {package_install_user, \"riakcs\"}.
 {package_install_group, \"riak\"}.
 {package_install_user_desc, \"Riak CS user\"}.
-{package_commands, {list, [[{name, \"riak-cs\"}], [{name, \"riak-cs-access\"}], [{name, \"riak-cs-gc\"}], [{name, \"riak-cs-storage\"}], [{name, \"riak-cs-stanchion\"}], [{name, \"riak-cs-cinfo\"}], [{name, \"riak-cs-debug\"}]]}}.
+{package_commands, {list, [[{name, \"riak-cs\"}], [{name, \"riak-cs-access\"}], [{name, \"riak-cs-gc\"}], [{name, \"riak-cs-storage\"}], [{name, \"riak-cs-stanchion\"}], [{name, \"riak-cs-cinfo\"}], [{name, \"riak-cs-debug\"}], [{name, \"riak-cs-multibag\"}]]}}.
```
Comment (kuenishi) [inline rebar.config.script]: This line will conflict against `riak-cs-admin` patch.

File: rel/files/riak-cs-multibag [hunk 2]
```diff
@@ -0,0 +1,27 @@
+#!/bin/sh
```
Comment (kuenishi) [inline rel/files/riak-cs-multibag:1]: This should also be adapted with `riak-cs-admin` style.

Comment (shino) [inline rel/files/riak-cs-multibag:1]: What do you mean by riak-cs-admin style?
If you mean that riak-cs-admin will include the calls of this script,
I'm afraid that it results in ugly error message like 'undef' in OSS package.

Comment (shino) [inline rel/files/riak-cs-multibag:1]: Current implementation (or my intention) is as follows:
- riak-cs-multibag is independent of riak-cs-admin (riak-cs-admin does not call it)
- `rebar generage` (or overlay setting in reltool.config) copies riak-cs-multibag script
  regardless oss or ee, but
- oss package does not include it and ee does (by pkg.vars.config generated by rebar.config.script).

(ugly? better way wanted :) )

File: src/riak_cs_block_server.erl [hunk 3]
```diff
@@ -87,16 +83,35 @@ start_link(RiakPid) ->
 %% MinWorkers. If the timeout occurs, this function
 %% could return an error, or the pids it has
 %% so far (which might be less than MinWorkers).
--spec start_block_servers(pid(), pos_integer()) -> [pid()].
-start_block_servers(RiakcPid, 1) ->
-    {ok, Pid} = start_link(RiakcPid),
-    [Pid];
-start_block_servers(RiakcPid, MaxNumServers) ->
-    case start_link() of
+-spec start_block_servers(lfs_manifest(), pid(), pos_integer()) -> [pid()].
```
Comment (kuenishi) [inline src/riak_cs_block_server.erl:86]: Maybe a good chance to [rename this function](https://github.com/basho/riak_cs/issues/828#issuecomment-39216738) .

File: src/riak_cs_bag_registrar.erl [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+process_specs() ->
+    (registar_module()):process_specs().
+
+pool_specs(Props) ->
+    (registar_module()):pool_specs(Props).
+
+%% Translate bag ID in buckets and manifests to pool name.
+-spec pool_name(pid(), pool_type(), term()) -> {ok, atom()} | {error, term()}.
+pool_name(MasterRiakc, PoolType, Target) ->
+    (registar_module()):pool_name(MasterRiakc, PoolType, Target).
```
Comment (kuenishi) [inline src/riak_cs_bag_registrar.erl:55]: Can we add a function like `get_pool_pid/3` as we repeat this pattern:

``` erlang
  {ok, PoolName} = pool_name(MasterPid, PoolType, Target),
  Pid = case PoolName of
             undefined -> MasterPid;
                     _ -> riak_cs_utils:riak_connection(MasterPid, PoolName)
             end,
   blahblahblah
```

Maybe we had a discussion on this in the past..

Comment (kuenishi) [inline src/riak_cs_bag_registrar.erl:55]: Or it can be callback function defined in `riak_cs_bag_registrar`

File: src/riak_cs_bag_registrar.erl [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+-callback process_specs() -> [term()].
+-callback pool_specs(proplists:proplist()) ->
+    [{atom(), {non_neg_integer(), non_neg_integer()}}].
+-callback pool_name(pid(), pool_type(), term()) -> {ok, atom()} | {error, term()}.
+-callback choose_bag_id(manifest | block, term()) -> bag_id().
+-callback set_bag_id_to_manifest(bag_id(), lfs_manifest()) -> lfs_manifest().
+-callback bag_id_from_manifest(lfs_manifest()) -> bag_id().
+-callback list_pool(pool_type()) -> [{Name::atom(), pool_type(), bag_id()}].
+-callback pool_status() -> [term()].
+-callback tab_info() -> term().
```
Comment (kuenishi) [inline src/riak_cs_bag_registrar.erl:41]: This callback seems never implemented. Maybe removed?

File: src/riak_cs_bag_registrar.erl [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+bag_id_from_manifest(Manifest) ->
+    (registar_module()):bag_id_from_manifest(Manifest).
+
+list_pool(PoolType) ->
+    (registar_module()):list_pool(PoolType).
+
+pool_status() ->
+    (registar_module()):list_pool().
+
+is_multibag_enabled() ->
```
Comment (kuenishi) [inline src/riak_cs_bag_registrar.erl:72]: This function should be in `riak_cs_config`.

**Comment by shino** [comment]
Close this in favor of #870

**Pull request closed without merging**